src/main/webapp/script.js

```diff
@@ -15,14 +15,182 @@
+        const titlesListElement = document.getElementById('titles-container');
+        titlesListElement.innerHTML = '';
+        var titlesList = fullList[0];
+        var contextsList = fullList[4];
+        var startDatesList = fullList[2];
+        var urlsList = fullList[8];
+        var locationsList = fullList[1];
+        var causesList = fullList[5];
+        var audiencesList = fullList[6];
+        var instructionsList = fullList[7];
+        for (i = 0; i < titlesList.length; i++) {
+            cardCount++;
+            titlesListElement.appendChild(createDivElement(cardCount, titlesList[i], startDatesList[i],
+             contextsList[i], urlsList[i], locationsList[i], causesList[i], audiencesList[i],
+             instructionsList[i]));
+        }
+        cardEnlargement(cardCount);
+    });
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
\ No newline at end of file
+function getBlobstoreUrl() {
+    fetch('/my-blobstore-url').then((response) => {
+        return response.text();
+    })
+    .then((imageUploadUrl) => {
+        const myForm = document.getElementById('newInitiativeForm');
+        myForm.action = imageUploadUrl;
+        myForm.classList.remove('hidden');
+      });
+}
+
+console.log(getBlobstoreUrl());
+
+function createDivElement(cardCount,initiativeTitle, initiativeStart,
+ initiativeContext, initiativeImage, initiativeLocation, initiativeCause, initiativeAudience, 
+ initiativeInstructions) {
+  const currentCard = document.createElement('div');
+  currentCard.id = "card" + cardCount;
+  const currStartDate = document.createElement('p');
+  const currTitle = document.createElement('p');
+  const currContext = document.createElement('p');
+  const currImage = document.createElement('img');
+  const currLocation = document.createElement('p');
+  const currCause = document.createElement('p');
+  const currAudience = document.createElement('p');
+  const currInstructions = document.createElement('p');
+  currLocation.innerText = initiativeLocation;
+  currLocation.id = "the-location" + "card" + cardCount;
+  currLocation.style.display = "none";
+  currAudience.innerText = initiativeAudience;
+  currAudience.id = "the-audience" + "card" + cardCount;
+  currAudience.style.display = "none";
+  currInstructions.innerText = initiativeInstructions;
+  currInstructions.id = "the-instructions" + "card" + cardCount;
+  currInstructions.style.display = "none";
+  currCause.innerText = initiativeCause;
+  currCause.id = "the-cause" + "card" + cardCount;
+  currCause.style.display = "none";
+  currImage.src = initiativeImage;
+  currImage.style.width = "85px";
+  currImage.style.height = "80px";
+  currImage.style.float = "left";
+  currImage.style.marginTop = "25px";
+  currImage.style.marginLeft = "15px";
+  currImage.style.marginRight = "13px";
```

Comment:
Is it possible to have all those styles in a separate CSS? Since you have the different IDs for each tag, I think instead of creating one id, you could perhaps have two class names for each tag (a generic name refers to whether lcoation or instruction... and another specific class name referring to that specific card) and use the generic name for styling.

Reply:
I see why you have these now, but maybe you could still keep these IDs, but also add class names to those tags to help with styling.

Reply:
yeah, this was very complicated since everything is kind of dynamically generated. Doing static CSS does not make sense since I as the website developer can never know the amount of cards there will be to make css tags to all of them before hand, I don't know if this makes sense to you.